tests/helper/helper_telemetry.go

```diff
@@ -1,39 +1,56 @@
 package helper
 
 import (
+	"context"
 	"encoding/json"
 	"io/ioutil"
 	"os"
 
 	_ "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
+	"github.com/redhat-developer/odo/pkg/config"
+	envcontext "github.com/redhat-developer/odo/pkg/config/context"
 	"github.com/redhat-developer/odo/pkg/preference"
 	"github.com/redhat-developer/odo/pkg/segment"
 )
 
+const (
+	DebugTelemetryFileEnv = "ODO_DEBUG_TELEMETRY_FILE"
+)
+
 func setDebugTelemetryFile(value string) error {
-	return os.Setenv(segment.DebugTelemetryFileEnv, value)
+	return os.Setenv(DebugTelemetryFileEnv, value)
 }
 
 // EnableTelemetryDebug creates a temp file to use for debugging telemetry.
 // it also sets up envs and cfg for the same
 func EnableTelemetryDebug() {
 	Expect(os.Setenv(segment.TrackingConsentEnv, "yes")).NotTo(HaveOccurred())
-	cfg, _ := preference.NewClient()
-	err := cfg.SetConfiguration(preference.ConsentTelemetrySetting, "true")
+
+	ctx := context.Background()
+	envConfig, err := config.GetConfiguration()
+	Expect(err).To(BeNil())
+	ctx = envcontext.WithEnvConfig(ctx, *envConfig)
+
+	cfg, _ := preference.NewClient(ctx)
+	err = cfg.SetConfiguration(preference.ConsentTelemetrySetting, "true")
 	Expect(err).To(BeNil())
 	tempFile, err := ioutil.TempFile("", "telemetry")
 	Expect(err).NotTo(HaveOccurred())
 	Expect(setDebugTelemetryFile(tempFile.Name())).NotTo(HaveOccurred())
 	Expect(tempFile.Close()).NotTo(HaveOccurred())
 }
 
+func GetDebugTelemetryFile() string {
+	return os.Getenv(DebugTelemetryFileEnv)
```

Comment:
Shouldn't this env var be obtained via envcontext or are you using the library since it's a test helper?

Reply:
In integration tests, we don't need to rely on the envcontext package